client/src/main/java/org/eclipse/hono/client/RequestResponseClient.java

```diff
@@ -34,4 +34,19 @@
      * @return {@code true} if this client can be used to exchange messages with the peer.
      */
     boolean isOpen();
+
+    /**
+     * Sets the period of time after which any requests are considered to have timed out.
+     * <p>
+     * The client will fail the result handler passed in to any of the operations if no response
+     * has been received from the peer after the given amount of time.
+     * <p>
+     * When setting this property to 0, requests do not time out at all. Note that this will
+     * allow for unanswered requests piling up in the client, which eventually may cause the
```

Comment:
..., which finally may cause

sounds better to me. WDYT?

Reply:
hmm, I would say that *eventually* in this context is the right word to use, as in *eventually consistent* ...

Reply:
Yes, not saying that this is wrong, "eventually may" read only a little too unprobable for me, while I think this is definitely something that should be understood.
Anyway, can live with your original one as well, just had a slight demand to strengthen the statement.